PmipCallStackFilter.cs

```diff
@@ -50,70 +56,78 @@ public static DkmStackWalkFrame PmipStackFrame(DkmStackContext stackContext, Dkm
             return frame;
         }
 
-        public static void RefreshStackData(string fileName)
+        public static void RefreshStackData(int pid)
         {
-            try
-            {
-                if (!File.Exists(fileName))
-                    return;
+            DirectoryInfo taskDirectory = new DirectoryInfo(Path.GetTempPath());
+            FileInfo[] taskFiles = taskDirectory.GetFiles("pmip_" + pid + "_*.txt");
+
+            if (taskFiles.Length < 1)
+                return;
+
+            Array.Sort(taskFiles, (a, b) => Path.GetFileNameWithoutExtension(a.Name).CompareTo(Path.GetFileNameWithoutExtension(b.Name)));
+            var fileName = taskFiles[taskFiles.Length - 1].FullName;
 
-                var fileInfo = new FileInfo(fileName);
-                if (fileInfo.Length == _previousFileLength)
-                    return;
+            if (_currentFile != fileName)
+            {
+                _fileStreamReader?.Dispose();
+                _fileStream?.Dispose();
+                _rangesSortedByIp.Clear();
 
-                var list = new List<Range>(10000);
-                using (var inStream = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                try
+                {
+                    _fileStream = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.ReadWrite | FileShare.Delete);
+                    _fileStreamReader = new StreamReader(_fileStream);
+                    _currentFile = fileName;
+                }
+                catch (Exception ex)
                 {
-                    using (var file = new StreamReader(inStream))
-                    {
-                        string line;
-                        while ((line = file.ReadLine()) != null)
-                        {
-                            const char delemiter = ';';
-                            var tokens = line.Split(delemiter);
-
-                            //should never happen, but lets be safe and not get array out of bounds if it does
-                            if (tokens.Length != 3)
-                                continue;
-
-                            var startip = tokens[0];
-                            var endip = tokens[1];
-                            var description = tokens[2];
-
-                            var startiplong = ulong.Parse(startip, NumberStyles.HexNumber);
-                            var endipint = ulong.Parse(endip, NumberStyles.HexNumber);
-
-                            list.Add(new Range() { Name = description, Start = startiplong, End = endipint });
-                        }
-                    }
+                    Console.WriteLine("Unable to read dumped pmip file: " + ex.Message);
```

Comment:
Does this go somewhere useful?

Reply:
No. I think theres a log function to print to the output window in VS...I'll take a look